prowler/providers/iac/iac_provider.py

```diff
@@ -0,0 +1,176 @@
+
+
+class IacProvider(Provider):
+    _type: str = "iac"
+    audit_metadata: Audit_Metadata
+
+    def __init__(
+        self,
+        scan_path: str = ".",
+        config_path: str = None,
+        config_content: dict = None,
+        fixer_config: dict = {},
+        mutelist_path: str = None,
+        mutelist_content: dict = None,
+    ):
+        logger.info("Instantiating IAC Provider...")
+
+        self.scan_path = scan_path
+        self.region = "global"
+        self.audited_account = "local-iac"
+        self._session = None
+        self._identity = "prowler"
+
+        self._audit_config = config_content if config_content else {}
+        self._fixer_config = fixer_config
+        self._mutelist = None
+
+        self.audit_metadata = Audit_Metadata(
+            provider=self._type,
+            account_id=self.audited_account,
+            account_name="iac",
+            region=self.region,
+            services_scanned=0,  # IAC doesn't use services
+            expected_checks=[],  # IAC doesn't use checks
+            completed_checks=0,  # IAC doesn't use checks
+            audit_progress=0,  # IAC doesn't use progress tracking
+        )
+
+        Provider.set_global_provider(self)
+
+    @property
+    def type(self):
+        return self._type
+
+    @property
+    def identity(self):
+        return self._identity
+
+    @property
+    def session(self):
+        return self._session
+
+    @property
+    def audit_config(self):
+        return self._audit_config
+
+    @property
+    def fixer_config(self):
+        return self._fixer_config
+
+    def setup_session(self):
+        """IAC provider doesn't need a session since it uses Checkov directly"""
+        return None
+
+    def run(self) -> List[CheckReportIAC]:
+        return self.run_scan(self.scan_path)
+
+    def run_scan(self, directory: str) -> List[CheckReportIAC]:
+        try:
+            logger.info(f"Running IaC scan on {directory}...")
+
+            # Run Checkov with JSON output
+            process = subprocess.run(
+                ["checkov", "-d", directory, "-o", "json"],
+                capture_output=True,
+                text=True,
+            )
+
+            try:
+                output = json.loads(process.stdout)
+                if not output:
+                    logger.warning("No findings returned from Checkov scan")
+                    return []
+            except json.JSONDecodeError:
+                logger.error("Failed to parse IaC scan output as JSON")
+                return []
+
+            reports = []
+
+            # Process failed checks
+            for finding in output:
+                failed_checks = finding.get("results", {}).get("failed_checks", [])
+                if not failed_checks:
+                    continue
+
+                for failed_check in failed_checks:
+                    # Get severity and ensure it's a valid string
+                    severity = finding.get("severity", "low")
```

Comment:
Maybe until we can fetch the real severity we should leave it as `None` o any other value that does not mislead importance of the finding..